src/quicksort.rs

```diff
@@ -17,41 +17,32 @@ fn quicksort<T: Ord>(slice: &mut [T]) {
     let (pivot, rest) = slice.split_first_mut().expect("slice is non-empty");
     let mut left = 0;
     let mut right = rest.len() - 1;
-    while left <= right {
+    // if right <= 0, it's done
```

Comment:
Maybe also point out here (this helped me) that if `right <= 0`, then `left == right`, so there can't be a need to swap.